### ISSUE TITLE
Weird error when passing guards to another thread

### ISSUE THREAD

**Issue by Ved-s** [Amanieu/parking_lot#399]
![](https://github.com/Amanieu/parking_lot/assets/53968411/f6acae7e-b3c2-4f11-b914-c4f9141a4cfa)
![](https://cdn.discordapp.com/attachments/986671988720414820/1146038419965886625/image.png)
Guards implement Send for some reason?

**Comment by bjorn3** [comment]
parking_lot's `RwLockWriteGuard` implements `Send` when the protected value implements it.

**Comment by Amanieu** [comment]
The error message with `std` is nicer because it uses an unstable feature to directly implement `!Send` on mutex guards. `parking_lot` uses a different method (see `GuardNoSend`) which effectively has the same effect.

**Comment by Ved-s** [comment]
what was the reason of doing that instead of !Send ?

**Comment by bjorn3** [comment]
`impl !Send for Type {}` is unstable, so parking_lot can't use it. The standard library is privileged in that it can use any unstable feature even on stable rustc.

**Comment by Ved-s** [comment]
Makes sense, though I've seen parking_lot use impl !Sync (ig it'sstable?)
